### ISSUE TITLE
compatability issues with installation of deepsea

### ISSUE THREAD

**Issue by cmlakhan** [kipoi/kipoi-veff#44]
I am trying to install Kipoi-Veff in conjunction with DeepSea and seem to be having problems.  I am using an AWS instance with the deep learning AMI.  I simply activate the python3 + cuda 10.0 environment and then install kipoi-veff using conda.  This seems to install kipoi and kipoi-veff but then when I then try to install DeepSea into the same environment

kipoi env install DeepSEA/predict --gpu

I am getting compatibility issues which eventually break things.

If I am trying to install everything from scratch, how would you recommend I install Kipoi, Kipoi-veff, Basenji, and DeepSea?

**Comment by haimasree** [comment]
Will you be interested in exploring docker containers for this purpose? 
I have the cpu versions of Basenji and DeepSea tested with this docker image 
```
docker pull haimasree/kipoi-docker:sharedpy3keras2
```
If you would like, I can create gpu friendly docker images as well.
If you need kipoi separately, 

```
docker pull haimasree/kipoi-docker:kipoi-base-env
```
I have not worked with kipoi-veff yet but as discussed I have started looking into it.

**Comment by cmlakhan** [comment]
Yeah that could be helpful.  I am using an HPC cluster at my institution so
I will have to see if they support docker.  Are you saying you have
DeepSea, Basenji, and Kipoi-Veff installed on a single docker instance?  I
have problems with Kipoi-veff interacts with the main kipoi installation.

Chirag

**Comment by haimasree** [comment]
Hi Chirag,
      Short answer: Sorry no. I tried installing kipoi-veff  with pip
install kipoi-veff inside the docker container
haimasree/kipoi-docker:sharedpy3keras2 (*cpu version*) and it seems to
install fine. I am attaching the conda environment if it is of any help.
This docker image is based on continuumio/miniconda3:4.8.2 which in turn I
think is based on debian. So, if your flavor of os is drastically different
then these libraries in the environment file will most likely not be
available. Maybe, some of the pinned package versions help?

Further, to elaborate, I have been working on making kipoi models more
robust for about 3/4 months. During which I created the docker images
containing *cpu only *conda environments covering the 29 model groups in
Kipoi. More information is available here -
https://github.com/haimasree/kipoi-containers. I have plans to containerize
the gpu version too shortly. I only started looking into kipoi-veff like
this week, We are planning to improve the usability of kipoi-veff to avoid
exactly the kind of problem you are facing right now. Hopefully it will get
resolved soon but I am afraid we are just in the initial planning phase :)

On another note - Clusters which do not allow docker but allows singularity
can use singularity to convert the docker containers to singularity
containers. See the "singularity support" section in and
https://github.com/haimasree/kipoi-containers
 and
https://github.com/haimasree/kipoi-containers/blob/main/build-singularity-container.sh#L24-L25

Thank you for reporting the issue. I am sorry I dont have a solution just
yet but I will be sure to post it if I have one.

Regards,
Haimasree
